server/API/REST.php

```diff
@@ -0,0 +1,170 @@
+<?php
+session_start();
+require_once "../vendor/autoload.php";
+
+header("Access-Control-Allow-Origin: *");
+header("Access-Control-Allow-Methods: PUT, GET, POST");
+header("Access-Control-Allow-Headers: Origin, X-Requested-With, Content-Type, Accept");
+header("Content-Type: application/json");
+
+define("API_PATH", $_SERVER["SCRIPT_NAME"]."/api");
+
+/* Turning warning and notices into exceptions */
+
+set_error_handler(function($errno, $errstr, $errfile, $errline) {
+    // error was suppressed with the @-operator
+    if (0 === error_reporting()) {
+        return false;
+    }
+
+    throw new ErrorException($errstr, 0, $errno, $errfile, $errline);
+});
+
+/* Utilities */
+
+if(!function_exists("check_login")){
+	/**
+	 * Check if a user is logged
+	 * 
+	 * @return bool true if is logged, false otherwise
+	 */
+	function check_login(){
+			$pdo = new PDO("sqlite:../db.sqlite");
+
+			$stmt = $pdo->prepare("SELECT ID, username, password, type FROM users WHERE ID = :userId");
+			$stmt->bindValue(":userId", $_SESSION["user_id"]);
+
+			if(!$stmt->execute()){
+				throw new Error($stmt->errorInfo(), $stmt->errorCode());
+			}
+
+			$user_data = $stmt->fetch();
+
+			return $_SESSION["nonce"] == md5(serialize($user_data));
+	}
+}
+
+
+/* Functions that implements endpoints */
+
+if(!function_exists("do_login")){
+	function do_login($vars){
+		try{
+			$pdo = new PDO("sqlite:../db.sqlite");
+
+			$stmt = $pdo->prepare("SELECT ID, username, password, type FROM users WHERE username = :username");
+			$stmt->bindValue(":username", $_POST["username"]);
+
+			if(!$stmt->execute()){
+				throw new Error($stmt->errorInfo(), $stmt->errorCode());
+			}
+
+			$user_data = $stmt->fetch();
+
+			if(!password_verify($_POST["password"], $user_data["password"])){
+				http_response_code(403);
+				echo json_encode(array('success' => false));
+				return;
+			}
+			else{
+				//TODO da trasformare in un vero nonce da salvare nel db
+				$nonce = md5(serialize($user_data));
+
+				$_SESSION['user_id'] = $user_data["ID"];
+				$_SESSION['nonce'] = $nonce;
+
+				echo json_encode(array("success" => true, "userId" => 
+				$user_data['ID'], 'type' => $user_data['type']));
+
+				return;
+			}
+		}
+		catch(Exception $e){
+			echo json_encode(array('success' => false, 'reason' => $e->getMessage()));
+		}
+	}
+}
+
+if(!function_exists('am_i_logged')){
+	function am_i_logged(){
+		if(!check_login()){
+			http_response_code(403);
```

Comment:
Not a client error, should be 200

Reply:
giving 403 as status code is a bit better in particular case because fetch will reject insted of resolving: the client code will be cleaner